frappe/email/bulk.py

```diff
@@ -185,19 +185,24 @@ def flush(from_test=False):
 		else:
 			break
 
-		frappe.db.sql("""update `tabBulk Email` set status='Sending' where name=%s""",
-			(email["name"],), auto_commit=auto_commit)
-		try:
-			if not from_test:
-				smtpserver.setup_email_account(email.reference_doctype)
-				smtpserver.sess.sendmail(email["sender"], email["recipient"], encode(email["message"]))
-
-			frappe.db.sql("""update `tabBulk Email` set status='Sent' where name=%s""",
+		if time_diff_in_seconds(None, email["creation"]) > 259200:
+			# expire mails older than 3 days
+			frappe.db.sql("""update `tabBulk Email` set status='Expired' where name=%s""",
+				(email["name"],), auto_commit=auto_commit)
```

Comment:
may be there could be single update query to expire any bulk email created 3 days ago and not sent?